contrib/utilities/generate_lapack_templates

```diff
@@ -0,0 +1,650 @@
+    parser.add_argument("--lapack-subroutines", type=str,
+                        help="file listing which LAPACK subroutines we should "
+                        "wrap. Defaults to 'lapack_templates_data/lapack_subroutines' "
+                        "in the current directory.")
+    parser.add_argument("--blas-subroutines", type=str,
+                        help="file listing which BLAS subroutines we should "
+                        "wrap. Defaults to 'lapack_templates_data/blas_subroutines' "
+                        "in the current directory.")
+    parser.add_argument("--fp-guarded-subroutines", type=str,
+                        help="Subroutines which should be wrapped with "
+                        "disabling and enabling of floating point exceptions. "
+                        "Defaults to 'lapack_templates_data/fp_subroutines' in "
+                        "the current directory.")
+    parser.add_argument("--output-file", type=str,
+                        help="File to which we write. Defaults to "
+                        "'lapack_templates.h' in the current directory.")
+    args = vars(parser.parse_args())
+
+    if args["lapack_root_directory"] is None:
+        raise ValueError("The path to the default LAPACK directory must be "
+                         "specified.")
+
+    if args["blas_subroutines"] is None:
+        args["blas_subroutines"] = "lapack_templates_data/blas_subroutines"
+    if args["lapack_subroutines"] is None:
+        args["lapack_subroutines"] = "lapack_templates_data/lapack_subroutines"
+    if args["fp_guarded_subroutines"] is None:
+        args["fp_guarded_subroutines"] = "lapack_templates_data/fp_subroutines"
+    if args["output_file"] is None:
+        args["output_file"] = "lapack_templates.h"
+    print(args)
+
+    with open(args["lapack_subroutines"]) as lapack_handle:
+        for line in lapack_handle:
+            lapack_subroutines.append(line.strip())
+    with open(args["blas_subroutines"]) as blas_handle:
+        for line in blas_handle:
+            blas_subroutines.append(line.strip())
+    with open(args["fp_guarded_subroutines"]) as fp_handle:
+        for line in fp_handle:
+            fp_guarded_subroutines.append(line.strip())
+
+    #
+    # Parse subroutines:
+    #
+    routines = list()
+    all_general_routines = blas_subroutines + lapack_subroutines
+    for [general_routines, routine_directory] in [
+            [lapack_subroutines, args["lapack_root_directory"] + "/SRC/"],
+            # dlamch and slamch are in the INSTALL directory
+            [lapack_subroutines, args["lapack_root_directory"] + "/INSTALL/"],
+            [blas_subroutines, args["lapack_root_directory"] + "/BLAS/SRC/"]]:
+        for general_routine in general_routines:
+            globs = glob.glob(routine_directory + "*" + general_routine + ".f")
+            for file_name in globs:
+                try:
+                    routine_name = osp.basename(file_name)[1:-2]
```

Comment:
did we have any non-`s`/`d`/`c`/`z` routines (some helper routines). This would not work here. But it should be easy to extend if needed by doing 
```
[lapack_subroutines, true, args["lapack_root_directory"] + "/SRC/"]
[helper_subroutines, false, args["lapack_root_directory"] + "/SOME/PATH/"]
```
and then based on the switch act differently.

Reply:
I did not find any routines that did not fit the classic naming scheme. There are only a few LAPACK functions that don't follow the pattern:

```
icmax1.f
ieeeck.f
ilaclc.f
ilaclr.f
iladiag.f
iladlc.f
iladlr.f
ilaenv2stage.f
ilaenv.f
ilaprec.f
ilaslc.f
ilaslr.f
ilatrans.f
ilauplo.f
ilazlc.f
ilazlr.f
iparam2stage.F
iparmq.f
izmax1.f
lsamen.f
xerbla_array.f
xerbla.f
```

and for BLAS

```
icamax.f
idamax.f
isamax.f
izamax.f
lsame.f
xerbla_array.f
xerbla.f
```

These are either helper functions, internal functions (`ilaenv.f` checks environment variables), or obscure (`ilaclc.f` scans a matrix to find its last non-zero column). I don't think we should bother supporting these now. At some point in the future we could extend the parser to do it with inheritance (some version would not have the 'print a template base case') if needed.